io.quarkus:quarkus-jacoco cannot import pom module dependecy

### Describe the bug

During my development I had the need to import a pom type submodule in my quarkus project, where I already had the quarkus-jacoco dependency.

parent/pom.xml
    
    <?xml version="1.0" encoding="ISO-8859-15"?>
    <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
        <modelVersion>4.0.0</modelVersion>
        <groupId>org.acme</groupId>
        <artifactId>parent</artifactId>
        <version>1.0.0-SNAPSHOT</version>
        <packaging>pom</packaging>
        <modules>
            <module>getting-started</module>
            <module>bug</module>
        </modules>
    </project>


bug/pom.xml

    <?xml version="1.0" encoding="ISO-8859-15"?>
    <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    
        <modelVersion>4.0.0</modelVersion>
    
        <parent>
            <groupId>org.acme</groupId>
            <artifactId>parent</artifactId>
            <version>1.0.0-SNAPSHOT</version>
            <relativePath>../pom.xml</relativePath>
        </parent>
    
        <artifactId>bug</artifactId>
        <packaging>pom</packaging>
    
        <name>bug</name>
    </project>


quarkus-getting-started/pom.xml
    
    <?xml version="1.0" encoding="UTF-8"?>
    <project xmlns="http://maven.apache.org/POM/4.0.0"
             xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
             xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    (...)
      <parent>
          <groupId>org.acme</groupId>
          <artifactId>parent</artifactId>
          <version>1.0.0-SNAPSHOT</version>
          <relativePath>../pom.xml</relativePath>
      </parent>
      (...)
      <dependencies>
            <!-- incompatible dependencies?! -->
            <dependency>
                <groupId>${quarkus.platform.group-id}</groupId>
                <artifactId>quarkus-jacoco</artifactId>
            </dependency>
            <dependency>
                <groupId>org.acme</groupId>
                <artifactId>bug</artifactId>
                <version>1.0.0-SNAPSHOT</version>
                <type>pom</type>
            </dependency>
      </dependencies>
      (...)
      </project>

### Expected behavior

If you remove this submodule "bug" or if you change the type to "jar" or if you remove quarkus-jacoco the tests work correctly.



### Actual behavior

In the testing phase with coverage I get the exception below:

    [INFO] -------------------------------------------------------
    [INFO]  T E S T S
    [INFO] -------------------------------------------------------
    [INFO] Running pt.segsocial.ssn.async.MetaendpointsTestIT
    [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.516 s <<< FAILURE! - in pt.segsocial.ssn.async.MetaendpointsTestIT
    [ERROR] pt.segsocial.ssn.async.MetaendpointsTestIT.shouldReturnOKWhenHealthEndpoint  Time elapsed: 0.01 s  <<< ERROR!
    java.lang.RuntimeException: 
    java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	    [error]: Build step io.quarkus.jacoco.deployment.JacocoProcessor#transformerBuildItem threw an exception: java.lang.NullPointerException
	    at io.quarkus.jacoco.deployment.JacocoProcessor.addProjectModule(JacocoProcessor.java:126)
	    at io.quarkus.jacoco.deployment.JacocoProcessor.transformerBuildItem(JacocoProcessor.java:113)
	    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	    at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	    at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
	    at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
	    at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	    at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
	    at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
	    at java.base/java.lang.Thread.run(Thread.java:834)
	    at org.jboss.threads.JBossThread.run(JBossThread.java:501)
    
	    at io.quarkus.test.junit.QuarkusTestExtension.throwBootFailureException(QuarkusTestExtension.java:625)
	    at io.quarkus.test.junit.QuarkusTestExtension.interceptTestClassConstructor(QuarkusTestExtension.java:696)
	    at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
	    at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	    at org.junit.jupiter.api.extension.InvocationInterceptor.interceptTestClassConstructor(InvocationInterceptor.java:73)
	    at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
	    at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	    at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	    at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	    at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	    at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:92)
	    at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:62)
	    at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeTestClassConstructor(ClassBasedTestDescriptor.java:363)
	    at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.instantiateTestClass(ClassBasedTestDescriptor.java:310)
	    at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.instantiateTestClass(ClassTestDescriptor.java:79)
	    at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.instantiateAndPostProcessTestInstance(ClassBasedTestDescriptor.java:286)
	    at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$testInstancesProvider$4(ClassBasedTestDescriptor.java:278)
	    at java.base/java.util.Optional.orElseGet(Optional.java:369)
	    at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$testInstancesProvider$5(ClassBasedTestDescriptor.java:277)
	    at org.junit.jupiter.engine.execution.TestInstancesProvider.getTestInstances(TestInstancesProvider.java:31)
	    at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$prepare$0(TestMethodTestDescriptor.java:105)
	    at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	    at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.prepare(TestMethodTestDescriptor.java:104)
	    at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.prepare(TestMethodTestDescriptor.java:68)
	    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$prepare$2(NodeTestTask.java:123)
	    at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	    at org.junit.platform.engine.support.hierarchical.NodeTestTask.prepare(NodeTestTask.java:123)
	    at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:90)
	    at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	    at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	    at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	    at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	    at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	    at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	    at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	    at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	    at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	    at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	    at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	    at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	    at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	    at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	    at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	    at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	    at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
	    at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:147)
	    at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:127)
	    at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:90)
	    at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:55)
	    at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:102)
	    at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:54)
	    at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	    at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	    at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	    at org.apache.maven.surefire.junitplatform.LazyLauncher.execute(LazyLauncher.java:50)
	    at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:184)
	    at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:148)
	    at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:122)
	    at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
	    at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	    at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
	    at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
    Caused by: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	    [error]: Build step io.quarkus.jacoco.deployment.JacocoProcessor#transformerBuildItem threw an exception: java.lang.NullPointerException
	    at io.quarkus.jacoco.deployment.JacocoProcessor.addProjectModule(JacocoProcessor.java:126)
	    at io.quarkus.jacoco.deployment.JacocoProcessor.transformerBuildItem(JacocoProcessor.java:113)
	    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	    at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	    at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
	    at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
	    at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	    at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
	    at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
	    at java.base/java.lang.Thread.run(Thread.java:834)
	    at org.jboss.threads.JBossThread.run(JBossThread.java:501)
    
	    at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:335)
	    at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:252)
	    at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:60)
	    at io.quarkus.test.junit.QuarkusTestExtension.doJavaStart(QuarkusTestExtension.java:222)
	    at io.quarkus.test.junit.QuarkusTestExtension.ensureStarted(QuarkusTestExtension.java:592)
	    at io.quarkus.test.junit.QuarkusTestExtension.beforeAll(QuarkusTestExtension.java:640)
	    at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllCallbacks$12(ClassBasedTestDescriptor.java:395)
	    at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	    at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAllCallbacks(ClassBasedTestDescriptor.java:395)
	    at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:211)
	    at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:84)
	    at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:148)
	    ... 37 more
    Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	    [error]: Build step io.quarkus.jacoco.deployment.JacocoProcessor#transformerBuildItem threw an exception: java.lang.NullPointerException
	    at io.quarkus.jacoco.deployment.JacocoProcessor.addProjectModule(JacocoProcessor.java:126)
	    at io.quarkus.jacoco.deployment.JacocoProcessor.transformerBuildItem(JacocoProcessor.java:113)
	    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	    at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	    at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
	    at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
	    at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	    at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
	    at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
	    at java.base/java.lang.Thread.run(Thread.java:834)
	    at org.jboss.threads.JBossThread.run(JBossThread.java:501)
    
	    at io.quarkus.builder.Execution.run(Execution.java:123)
	    at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:79)
	    at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:160)
	    at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:331)
	    ... 48 more
    Caused by: java.lang.NullPointerException
	    at io.quarkus.jacoco.deployment.JacocoProcessor.addProjectModule(JacocoProcessor.java:126)
	    at io.quarkus.jacoco.deployment.JacocoProcessor.transformerBuildItem(JacocoProcessor.java:113)
	    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	    at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	    at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
	    at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
	    at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	    at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
	    at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
	    at java.base/java.lang.Thread.run(Thread.java:834)
	    at org.jboss.threads.JBossThread.run(JBossThread.java:501)
    
    [INFO] 
    [INFO] Results:
    [INFO] 
    [ERROR] Errors: 
    [ERROR]   MetaendpointsTestIT.shouldReturnOKWhenHealthEndpoint » Runtime java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	    [error]: Build step io.quarkus.jacoco.deployment.JacocoProcessor#transformerBuildItem threw an exception: java.lang.NullPointerException
	    at io.quarkus.jacoco.deployment.JacocoProcessor.addProjectModule(JacocoProcessor.java:126)
	    at io.quarkus.jacoco.deployment.JacocoProcessor.transformerBuildItem(JacocoProcessor.java:113)
	    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	    at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	    at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
	    at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
	    at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	    at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
	    at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
	    at java.base/java.lang.Thread.run(Thread.java:834)
	    at org.jboss.threads.JBossThread.run(JBossThread.java:501)
    
    [INFO] 
    [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0

### How to Reproduce?

I created a small PoC:

[quarkus-bug.zip](https://github.com/quarkusio/quarkus/files/11615910/quarkus-bug.zip)


### Output of `uname -a` or `ver`

Linux neotolemus 6.3.3-100.fc37.x86_64 #1 SMP PREEMPT_DYNAMIC Wed May 17 15:02:48 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "17.0.7" 2023-04-18 OpenJDK Runtime Environment (Red_Hat-17.0.7.0.7-4.fc37) (build 17.0.7+7) OpenJDK 64-Bit Server VM (Red_Hat-17.0.7.0.7-4.fc37) (build 17.0.7+7, mixed mode, sharing)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

3.0.3.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.5 (Red Hat 3.8.5-3)

### Additional information

_No response_

--- a/test-framework/jacoco/deployment/src/main/java/io/quarkus/jacoco/deployment/JacocoProcessor.java
+++ b/test-framework/jacoco/deployment/src/main/java/io/quarkus/jacoco/deployment/JacocoProcessor.java
@@ -123,6 +123,9 @@ public class JacocoProcessor {
     private void addProjectModule(ResolvedDependency module, JacocoConfig config, ReportInfo info, String includes,
             String excludes, Set<String> classes, Set<String> sources) throws Exception {
         info.savedData.add(new File(module.getWorkspaceModule().getBuildDir(), config.dataFile).getAbsolutePath());
+        if (module.getSources() == null) {
+            return;
+        }
         for (SourceDir src : module.getSources().getSourceDirs()) {
             for (Path p : src.getSourceTree().getRoots()) {
                 sources.add(p.toAbsolutePath().toString());

